airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/java/io/airbyte/integrations/base/destination/typing_deduping/DefaultTyperDeduper.java

```diff
@@ -122,31 +127,49 @@ public void prepareTables() throws Exception {
     overwriteStreamsWithTmpTable = ConcurrentHashMap.newKeySet();
     LOGGER.info("Preparing tables");
 
+    // This is intentionally not done in parallel to avoid rate limits in some destinations.
     prepareSchemas(parsedCatalog);
-    final Set<CompletableFuture<Optional<Exception>>> prepareTablesTasks = new HashSet<>();
-    for (final StreamConfig stream : parsedCatalog.streams()) {
-      prepareTablesTasks.add(prepareTablesFuture(stream));
-    }
-    CompletableFuture.allOf(prepareTablesTasks.toArray(CompletableFuture[]::new)).join();
-    reduceExceptions(prepareTablesTasks, "The following exceptions were thrown attempting to prepare tables:\n");
+
+    // TODO: Either the migrations run the soft reset and create v2 tables or the actual prepare tables.
+    // unify the logic
+    // with current state of raw tables & final tables. This is done first before gather initial state
+    // to avoid recreating
+    // final tables later again.
+    final List<Either<? extends Exception, Void>> runMigrationsResult =
```

Comment:
there's some interesting pr dependencies here... my pr https://github.com/airbytehq/airbyte/pull/35376 (diff is gross, needs rebase) is also moving moving "run migrations" to a thing that happens before prepareTables. In retrospect I wonder if it would made sense for you to base this branch on top of mine 😅 

(... I don't particularly want to do that rebasing now that we've both written so many commits though)